test/hotel_system_test.rb

```diff
@@ -0,0 +1,179 @@
+require_relative 'test_helper'
+
+describe "HotelSystem class" do
+  
+  describe "initializer" do
+    before do
+      @system = Hotel::HotelSystem.new
+    end
+    
+    it "is an instance of HotelSystem class" do
+      expect(@system).must_be_kind_of Hotel::HotelSystem
+    end
+    
+    
+    it "establishes list of 20 rooms when instantiated" do
+      expect(@system.rooms.length).must_equal 20
+    end
+    
+    it "creates instance of room in room list" do
+      expect(@system.rooms.first).must_be_instance_of Hotel::Room
+      expect(@system.rooms.first.id).must_equal 1
+      expect(@system.rooms.last).must_be_instance_of Hotel::Room
+    end
+  end
+  
+  describe "make_reservation" do
+    before do
+      @system = Hotel::HotelSystem.new
+    end
+    
+    it "adds an instance of reservation to reservation list" do
+      expect(@system.reservations.length).must_equal 0
+      
+      @system.make_reservation(Date.new(2019,9,1), Date.new(2019,9,3))
+      
+      expect(@system.reservations.length).must_equal 1
+      expect(@system.reservations.last.start_date).must_equal Date.new(2019,9,1)
+    end
+    
+    it "adds reservation to 'dates_reserved' variable for the assigned room" do
+      expect(@system.rooms[0].dates_reserved).must_equal []
+      
+      @system.make_reservation(Date.new(2019,9,1), Date.new(2019,9,3))
```

Comment:
Consider storing the dates your testing in variables to make the relationships you're testing clearer and more flexible.

```suggestion
      start_date = Date.new(2019,9,1)
      end_date = start_date + 2
```